packages/core-app-api/src/lib/AuthSessionManager/RefreshingAuthSessionManager.ts

```diff
@@ -83,10 +98,17 @@ export class RefreshingAuthSessionManager<T> implements SessionManager<T> {
         }
         return refreshedSession;
       } catch (error) {
-        if (options.optional) {
-          return undefined;
+        try {
+          return createSession();
+        } catch (err) {
+          if (options.optional) {
+            return undefined;
+          }
+          throw new Error(
+            `Cannot create a new session after failing to refresh the session. Session refresh error: ${error}`,
+            { cause: err },
+          );
         }
-        throw error;
```

Comment:
I think it might be enough to just remove this `throw` to get the logic we want? and we might as well remove the optional check too as that's further down anyway. Essentially just making this the same as the empty catch down on :107

I do think we may want one more addition though unless it's handled elsewhere which is to `this.stateTracker.setSignedIn(false)` if we fail refresh and re-creation.

Reply:
agree, effectively call `signOut()`